src/curves/ed25519Bip32.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  stringToBytes,
+  bytesToHex,
+  hexToBigInt,
+  remove0x,
+} from '@metamask/utils';
+import { mod } from '@noble/curves/abstract/modular';
+import { ed25519 } from '@noble/curves/ed25519';
+
+export const curve = ed25519.CURVE;
+
+/**
+ * Named after whitepaper: BIP32-Ed25519 Hierarchical Deterministic Keys over a Non-linear Keyspace
+ * https://input-output-hk.github.io/adrestia/static/Ed25519_BIP.pdf
+ * "vanilla" "ed25519" curve follows SLIP10: https://tezos.stackexchange.com/questions/2837/can-i-use-bip32-hd-key-pairs-to-derive-ed25519-addresses
+ * note that that the important difference of the "bip32" version is that it allows unhardened key derivation
+ */
+export const name = 'ed25519Bip32';
+
+// Secret is empty string if not provided by user
+export const secret = stringToBytes('');
+
+/**
+ * Always returns true.
+ * For root node derivation, https://github.com/cardano-foundation/CIPs/blob/09d7d8ee1bd64f7e6b20b5a6cae088039dce00cb/CIP-0003/Icarus.md does not mention any cases when the private key is invalid.
+ * For child node derivation, https://input-output-hk.github.io/adrestia/static/Ed25519_BIP.pdf does not mention any cases when the private key is invalid.
+ *
+ * @param _privateKey - The private key to check.
+ * @returns True.
+ */
+export const isValidPrivateKey = (_privateKey: Uint8Array | string | bigint) =>
+  true;
+
+export const deriveUnhardenedKeys = true;
+
+export const publicKeyLength = 32;
+
+const bytesToNumberLE = (bytes: Uint8Array): bigint => {
+  return hexToBigInt(bytesToHex(Uint8Array.from(bytes).reverse()));
+};
+
+const modN = (a: bigint) => {
+  return mod(a, curve.n);
+};
+
+// Little-endian SHA512 with modulo n
+const modNLE = (hash: Uint8Array): bigint => {
+  return modN(bytesToNumberLE(hash));
+};
+
+const multiplyWithBase = (key: Uint8Array): Uint8Array => {
```

Comment:
this may look pretty arbitrary to a casual reader and they may legitimately wonder where did this come from - is there any reference implementation we can point at here?